timemachine/cpp/src/kernels/k_integrator.cuh

```diff
@@ -1,47 +1,52 @@
 #include "k_fixed_point.cuh"
 
-template <typename RealType>
-__global__ void update_forward_baoab(
+template <typename RealType, int D>
+__global__ void k_update_forward_baoab(
     const int N,
-    const int D,
     const RealType ca,
-    const unsigned int *__restrict__ idxs, // N
-    const RealType *__restrict__ cbs,      // N
-    const RealType *__restrict__ ccs,      // N
-    const RealType *__restrict__ noise,    // N x 3
-    RealType *__restrict__ x_t,
-    RealType *__restrict__ v_t,
-    const unsigned long long *__restrict__ du_dx,
+    const unsigned int *__restrict__ idxs,        // N
+    const RealType *__restrict__ cbs,             // N
+    const RealType *__restrict__ ccs,             // N
+    const RealType *__restrict__ noise,           // N x 3
+    RealType *__restrict__ x_t,                   // N x 3
+    RealType *__restrict__ v_t,                   // N x 3
+    const unsigned long long *__restrict__ du_dx, // N x 3
     const RealType dt) {
+    static_assert(D == 3);
 
     int kernel_idx = blockIdx.x * blockDim.x + threadIdx.x;
     if (kernel_idx >= N) {
         return;
     }
-    int atom_idx;
-    if (idxs) {
-        atom_idx = idxs[kernel_idx];
-    } else {
-        atom_idx = kernel_idx;
-    }
+    int atom_idx = idxs == nullptr ? kernel_idx : idxs[kernel_idx];
```

Comment:
parentheses for my sanity please

Reply:
Added in [527921d](https://github.com/proteneer/timemachine/pull/1109/commits/527921d972d990afcec783e287d1b9d62533970c)